src/Build/BuildCheck/Infrastructure/BuildCheckBuildEventHandler.cs

```diff
@@ -45,12 +50,38 @@ internal class BuildCheckBuildEventHandler
 
     public void HandleBuildEvent(BuildEventArgs e)
     {
+        if (
+            e.GetType() != typeof(BuildSubmissionStartedEventArgs) &&
+            e.BuildEventContext is not null &&
+            _projectRestore.TryGetValue(e.BuildEventContext.SubmissionId, out bool isRestoring) &&
+            isRestoring)
+        {
+            return;
+        }
+
         if (_eventHandlers.TryGetValue(e.GetType(), out Action<BuildEventArgs>? handler))
         {
             handler(e);
         }
     }
 
+    private void HandleBuildSubmissionStartedEvent(BuildSubmissionStartedEventArgs eventArgs)
+    {
+        if (_projectRestore.TryGetValue(eventArgs.SubmissionId, out bool isRestoring))
+        {
+            if (isRestoring)
```

Comment:
nit: The second condition could be merged to the parent one

Reply:
There is also `else` below so I am not quite sure about it. 
This code seems to be overcomplicated to me, while it is correct. Given that the `BuildSubmissionStartedEventArgs` is sent once per submission, how this submission could be in the dictionary already? Or I am wrong, and it could be sent more times?

Reply:
Yes, it was overcomplicated as I was still thinking about the prior implementation we had. I simplified for the restoring to just be a bool to check instead of a dictionary